website/src/hooks/tasks/useGenericTaskAPI.tsx

```diff
@@ -1,26 +1,43 @@
 import { useState } from "react";
 import { get, post } from "src/lib/api";
-import { BaseTask, TaskResponse, TaskType as TaskTypeEnum } from "src/types/Task";
+import { TaskApiHook } from "src/types/Hooks";
+import { BaseTask, TaskAvailableResponse, TaskResponse, TaskType as TaskTypeEnum } from "src/types/Task";
 import useSWRImmutable from "swr/immutable";
 import useSWRMutation from "swr/mutation";
 
-export const useGenericTaskAPI = <TaskType extends BaseTask>(taskType: TaskTypeEnum) => {
-  type ConcreteTaskResponse = TaskResponse<TaskType>;
+export const useGenericTaskAPI = <TaskType extends BaseTask>(taskType: TaskTypeEnum): TaskApiHook<TaskType> => {
+  const [response, setReponse] = useState<TaskResponse<TaskType>>({ status: "AWAITING_INITIAL" });
+  const [isLoading, setIsLoading] = useState(true);
 
-  const [tasks, setTasks] = useState<ConcreteTaskResponse[]>([]);
+  const { mutate: requestNewTask } = useSWRImmutable<TaskAvailableResponse<TaskType>>(
+    "/api/new_task/" + taskType,
+    get,
+    {
+      onSuccess: (response) => {
+        setIsLoading(false);
+        setReponse({ status: "AVAILABLE", ...response });
```

Comment:
I'm inclined to think that if we get a `200` from the API the task should be assumed available.